src/analyses/baseInvariant.ml

```diff
@@ -786,8 +786,14 @@ struct
         | CastE ((TEnum ({ekind = ik; _ }, _)) as t, e), Int c -> (* Can only meet the t part of an Lval in e with c (unless we meet with all overflow possibilities)! Since there is no good way to do this, we only continue if e has no values outside of t. *)
           (match eval e st with
            | Int i ->
-             if ID.leq i (ID.cast_to ik i) then
-               match unrollType (Cilfacade.typeOf e) with
+             let cinterval = ID.cast_to ik i in
+             let cl, cu = ID.minimal cinterval, ID.maximal cinterval in
+             let l, u = ID.minimal i, ID.maximal i in
+             (match cl,cu,l,u with
+              | Some cl, Some cu, Some l, Some u when Z.leq cl l && Z.leq u cu ->
+                (* Checking that the maximal and minimal values still fit instead of checking leq which may fail if one domain is *)
+                (* imprecise, e.g., DefExc *)
+                (match unrollType (Cilfacade.typeOf e) with
                | TInt(ik_e, _)
```

Comment:
Not sure why the indentation CI job gets skipped, but it looks screwed up now.

Reply:
I'll have a look. Actually, I am trying to pull this out and replace `VD.is_safe_cast: typ -> typ-> bool` with 

- `VD.is_staticlaly_safe_cast` (same as before)
- `VD.is_dynamically_safe_cast: typ -> typ -> VD.t -> bool` taking the current values into account.

I think there's probably more places where this would be the right thing to do, e.g., here:

https://github.com/goblint/analyzer/blob/092eed37b8803f0f1886d0cd83ada4f204c150a4/src/analyses/baseInvariant.ml#L210-L240

Reply:
The comment there seems suspicious, as `ID.cast_to_ik v = v` does not imply `VD.is_safe_cast t1 (Cilfacade.typeOfLval x)` and the reverse is also not true.